plugins/plugin-testing-java/plugin-testing-testng/che-plugin-testing-testng-runtime/src/main/java/org/testng/TestingMessageHelper.java

```diff
@@ -0,0 +1,268 @@
+     * @param out
+     *         output stream
+     * @param params
+     *         special parameters which can contain the name of failed test, error message
+     */
+    public static void testFailed(PrintStream out, Map<String, String> params) {
+
+        List<Pair> attributes = new ArrayList<>(params.size());
+        for (String key : params.keySet()) {
+            attributes.add(new Pair(key, escape(params.get(key))));
+        }
+        out.println(create("testFailed", attributes));
+    }
+
+    private static String create(String name, Pair... attributes) {
+        List<Pair> pairList = null;
+        if (attributes != null) {
+            pairList = Arrays.asList(attributes);
+        }
+        return create(name, pairList);
+    }
+
+    private static String create(String name, List<Pair> attributes) {
+        StringBuilder builder = new StringBuilder("@@<{\"name\":");
+        builder.append('"').append(name).append('"');
+        if (attributes != null) {
+            builder.append(", \"attributes\":{");
+            StringJoiner joiner = new StringJoiner(", ");
+            for (Pair attribute : attributes) {
+                joiner.add("\"" + attribute.first + "\":\"" + attribute.second + "\"");
+            }
+            builder.append(joiner.toString());
+            builder.append("}");
+        }
+
+        builder.append("}>");
+        return builder.toString();
+    }
+
+    private static String escape(String str) {
+        if (str == null) {
+            return null;
+        }
+
+        int escapeLength = calculateEscapedStringLength(str);
+        if (escapeLength == str.length()) {
+            return str;
+        }
+
+        char[] chars = new char[escapeLength];
+        int currentOffset = 0;
+        for (int i = 0; i < str.length(); i++) {
+            char c = str.charAt(i);
+            char escape = escapeChar(c);
+            if (escape != 0) {
+                chars[currentOffset++] = ESCAPE_SEPARATOR;
+                chars[currentOffset++] = escape;
+            } else {
+                chars[currentOffset++] = c;
+            }
+        }
+
+        return new String(chars);
+    }
+
+    private static int calculateEscapedStringLength(String string) {
+        int result = 0;
+        for (int i = 0; i < string.length(); i++) {
+            char c = string.charAt(i);
+            if (escapeChar(c) != 0) {
+                result += 2;
+            } else {
+                result++;
+            }
+        }
+        return result;
+    }
+
+
+    private static char escapeChar(char c) {
+        switch (c) {
+            case '\n':
+                return 'n';
+            case '\r':
+                return 'r';
+            case '\u0085':
+                return 'x';
+            case '\u2028':
+                return 'l';
+            case '\u2029':
+                return 'p';
+            default:
+                return 0;
+        }
+    }
+
+    private static class Pair {
```

Comment:
I don't sink so, this is runtime part of TestNG JVM, I want to keep less external dependencies inside this module as possible